### PR TITLE
Arch arm mpu assert additions and fixes

### PR DESCRIPTION
**Pull request by ioannisg**
Addresses #7384.
Addresses an additional bug related to region out-of-bounds configuration (see relevant comment).
Adds some minor fixes to the code and comment enhancements.

### CODE REVIEW

File: arch/arm/core/cortex_m/mpu/arm_mpu.c [hunk 1]
```diff
@@ -273,8 +273,7 @@ void arm_core_mpu_configure(u8_t type, u32_t base, u32_t size)
 	u32_t region_index = _get_region_index_by_type(type);
 	u32_t region_attr = _get_region_attr_by_type(type, size);
 
-	/* ARM MPU supports up to 16 Regions */
-	if (region_index > _get_num_regions()) {
```
Comment (ioannisg) [inline arch/arm/core/cortex_m/mpu/arm_mpu.c]: This looks like an out-of-bounds sanity-check bug, @agross-linaro, am I correct?

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
I think this code does too much.

It's sufficient to just make assertion checks, not try to prevent the system from crashing if assertions are off and people feed garbage data to the MPU driver. We only do this kind of rigorous runtime checking for drivers that we expose to user mode.

**Review by tyrantlizard77 — APPROVED** [review]
I think this is fine for 1.12

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

File: arch/arm/core/cortex_m/mpu/arm_mpu.c [hunk 2]
```diff
@@ -267,11 +302,14 @@ void arm_core_mpu_configure_user_context(struct k_thread *thread)
 #if defined(CONFIG_APPLICATION_MEMORY)
 	/* configure app data portion */
 	index = _get_region_index_by_type(THREAD_APP_DATA_REGION);
-	size = (u32_t)&__app_ram_end - (u32_t)&__app_ram_start;
-	region_attr = _get_region_attr_by_type(THREAD_APP_DATA_REGION, size);
-	if (size > 0)
-		_region_init(index, (u32_t)&__app_ram_start, region_attr);
-#endif
+	if (index < _get_num_regions())
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply arch/arm/core/cortex_m/mpu/arm_mpu.c]: Thanks!

**Review by tyrantlizard77 — APPROVED** [review]

File: arch/arm/core/cortex_m/mpu/arm_mpu.c [hunk 3]
```diff
@@ -143,7 +143,7 @@ static inline u32_t _get_region_index_by_type(u32_t type)
 }
 
 /**
- * This internal function check if region is enabled or not
+ * This internal function checks if region is enabled or not.
```
Comment (MaureenHelm) [inline arch/arm/core/cortex_m/mpu/arm_mpu.c]: Can you please make the same comment fixes in the nxp mpu?

**Review by MaureenHelm — COMMENTED** [review]
LGTM. Some of the comments you're fixing are also in the nxp mpu, could you please fix those too?

File: arch/arm/core/cortex_m/mpu/arm_mpu.c (hunk 3 above)
Comment (ioannisg) [reply arch/arm/core/cortex_m/mpu/arm_mpu.c]: Will have a look now.

**Comment by ioannisg** [comment]
@MaureenHelm  I pushed an additional commit that implements the comment&style fixes in NXP_MPU.

**Review by nashif — APPROVED** [review]

**Review by agross-oss — APPROVED** [review]

**Pull request merged**